src/app/admin/components/list-of-users/list-of-users.component.ts

```diff
@@ -1,48 +1,39 @@
-import { Component, OnInit } from '@angular/core';
-import { Observable, of, tap } from 'rxjs';
-import { User } from '../../../../shared/model/news.model';
+import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
+import { Observable, of, switchMap, tap } from 'rxjs';
 import { AuthenticationService } from '../../../../shared/services/authentication.service';
 import { MatTableDataSource } from '@angular/material/table';
+import { User } from '../../../../shared/model/user.model';
+import { DeletingUserComponent } from '../deleting-user/deleting-user.component';
+import { EditingUserDataComponent } from '../editing-user-data/editing-user-data.component';
+import { MatDialog } from '@angular/material/dialog';
+import { MatSort } from '@angular/material/sort';
+import { MatPaginator } from '@angular/material/paginator';
 
-export interface PeriodicElement {
-  name: string;
-  position: number;
-  weight: number;
-  symbol: string;
-}
-
-// const ELEMENT_DATA: PeriodicElement[] = [
-//   { position: 1, name: 'Hydrogen', weight: 1.0079, symbol: 'H' },
-//   { position: 2, name: 'Helium', weight: 4.0026, symbol: 'He' },
-//   { position: 3, name: 'Lithium', weight: 6.941, symbol: 'Li' },
-//   { position: 4, name: 'Beryllium', weight: 9.0122, symbol: 'Be' },
-//   { position: 5, name: 'Boron', weight: 10.811, symbol: 'B' },
-//   { position: 6, name: 'Carbon', weight: 12.0107, symbol: 'C' },
-//   { position: 7, name: 'Nitrogen', weight: 14.0067, symbol: 'N' },
-//   { position: 8, name: 'Oxygen', weight: 15.9994, symbol: 'O' },
-//   { position: 9, name: 'Fluorine', weight: 18.9984, symbol: 'F' },
-//   { position: 10, name: 'Neon', weight: 20.1797, symbol: 'Ne' },
-// ];
-
-/**
- * @title Basic use of `<table mat-table>`
- */
 @Component({
   selector: 'app-list-of-users',
   templateUrl: './list-of-users.component.html',
   styleUrls: ['./list-of-users.component.scss'],
 })
-export class ListOfUsersComponent implements OnInit {
-  displayedColumns: string[] = ['id', 'username', 'role', 'password'];
-  dataSource = new MatTableDataSource();
+export class ListOfUsersComponent implements OnInit, AfterViewInit {
+  displayedColumns: string[] = ['id', 'username', 'role', 'password', 'symbol'];
+  dataSource = new MatTableDataSource([]);
   userList$: Observable<User[]> = of([]);
+  dataToDisplay = [this.userList$];
   notAdmin = false;
 
-  constructor(private auth: AuthenticationService) {}
+  @ViewChild(MatPaginator) paginator!: MatPaginator;
+  @ViewChild(MatSort) sort!: MatSort;
+
+  constructor(private auth: AuthenticationService, public dialog: MatDialog) {
+    this.auth.getList().pipe(
+      tap(data => {
+        console.log(this.dataSource.data);
+      })
+    );
+  }
 
   ngOnInit() {
     this.loadData().subscribe();
-    this.getElementData();
   }
 
   loadData() {
```

Comment:
method for nothing

Reply:
sorry, wrong marker
i mean `loadData(){}` - it is the method for nothing
it is empty